Source/Provers/SMTLib/ProverInterface.cs

```diff
@@ -111,7 +110,7 @@ private void SetupAxiomBuilder(VCExpressionGenerator gen)
           AxBuilder.Setup();
           break;
         case CommandLineOptions.TypeEncoding.Monomorphic:
-          AxBuilder = new TypeAxiomBuilderPremisses(gen);
+          AxBuilder = null;
```

Comment:
Can you provide some context for the effect of this change?  For example, why is it safe to make this change?

Reply:
TypeAxiomBuilders are used to generate type axioms for polymorphic programs. Now, there used to be code in Boogie, which I removed in this pull request, that, to the best of my understanding, attempted to support polymorphic programs but have monomorphic arrays. This was controlled using Monomorphize and MonomorphicArrays flags.

Once we decided not to support that functionality, and note that when I tried it out it did not work in the first place, there was no need to be creating an axiom builder here for monomorphic programs.